cmd/exec/operations.go

```diff
@@ -450,40 +450,59 @@ func executeEvent(url string) (string, error) {
 	}
 
 	defer inputDataOsFile.Close()
-	byteResult, _ := ioutil.ReadAll(inputDataOsFile)
-	var event map[string]interface{}
-	json.Unmarshal([]byte(byteResult), &event)
+	byteResult, err := ioutil.ReadAll(inputDataOsFile)
+
+	if err != nil {
+		return "", err
+	}
 
-	// Read input data as bytes.buffer
-	inputData, err := safeLoadFile(localAbsPath)
+	var event map[string]interface{}
+	err = json.Unmarshal([]byte(byteResult), &event)
 
 	if err != nil {
-		log.Fatalf("Failed to load input file: %v", err)
+		return "", err
 	}
 
-	if inputData.Len() == 0 {
-		return event["id"].(string), errors.New("empty file ")
+	//fill or overwrite inputData if necessary
+	if Id != "" {
+		event["id"] = Id
+	}
+	if Source != "" {
+		event["source"] = Source
+	}
+	if Type != "" {
+		event["type"] = Type
 	}
-	if inputType != "application/json" {
-		return event["id"].(string), errors.New("filtetype not json")
+	if Specversion != "" {
+		event["specversion"] = Specversion
 	}
 
-	body := strings.NewReader(inputData.String())
+	if len(event) == 0 {
+		return "", errors.New("empty file ")
+	}
+
+	eventBody, err := json.Marshal(event)
+
+	if err != nil {
+		return "", err
+	}
+
+	body := strings.NewReader(string(eventBody))
```

Comment:
You probably want to use `bytes.NewReader` here. That way you don't need to cast it to a string.